src/App.tsx

```
@@ -1,27 +1,58 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
+import { Good } from './types/Good';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+  const [goodsViewMode, setGoodsViewMode] = useState<'all' | '5first' | 'red'>(
+    'all',
+  );
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+  useEffect(() => {
+    const loader =
+      goodsViewMode === 'all'
+        ? getAll
+        : goodsViewMode === '5first'
+          ? get5First
+          : getRedGoods;
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+    loader()
+      .then(setGoods)
+      .catch(() => setGoods([]));
```

Your error handling here prevents the app from crashing, which is good. To enhance the user experience, consider adding dedicated loading and error states. This would allow you to display a loading spinner while fetching data and show a user-friendly error message if the API call fails, rather than just showing an empty list.